### PR TITLE
crimson: add object_info, log support to crimson-store-nbd, fix lba btree bugs

### PR DESCRIPTION
**Pull request by athanatos**
Should address https://tracker.ceph.com/issues/52434 and https://tracker.ceph.com/issues/52435 as noted in https://github.com/ceph/ceph/pull/42901

### CODE REVIEW

**Comment by athanatos** [comment]
make check failure seems to have been due to  axboe/fio.git head being updated, added a commit pointing to new head.

**Review by liu-chunmei — APPROVED** [review]
LGTM, passing my test, fio run over 40min, iodepth=64, numjobs=4, 4K randwrite.

File: cmake/modules/BuildFIO.cmake [hunk 1]
```diff
@@ -16,7 +16,7 @@ function(build_fio)
     GIT_REPOSITORY "https://github.com/axboe/fio.git"
     GIT_CONFIG advice.detachedHead=false
     GIT_SHALLOW 1
-    GIT_TAG "7f02f21f53ccd5e2448086f8e9015489693dd2dc"
+    GIT_TAG "26976a134e44f583f91e05df95ef8ec5a9cc968d"
```
Comment (tchaikov) [inline cmake/modules/BuildFIO.cmake:19]: should have been fixed by #42955 .

File: src/crimson/tools/store_nbd/fs_driver.cc [hunk 2]
```diff
@@ -13,16 +15,96 @@ coll_t get_coll(unsigned num) {
...
+    0,
+    ghobject_t::NO_GEN);
+}
+
+std::string make_log_key(
+  unsigned i)
+{
+  std::stringstream ss;
+  ss << "log_entry_" << i;
+  return ss.str();
```
Comment (tchaikov) [inline src/crimson/tools/store_nbd/fs_driver.cc:35]: in future, i'd suggest use `fmt::format("log_entry_{}", i)` to implement this.

**Comment by liu-chunmei** [comment]
@athanatos  still got the error: 
ERROR 2021-08-27 22:49:31,144 [shard 0] seastore - add_pin: unable to add btree_ range_pin_t(begin=0, end=18446744073709551615, depth=3, extent=0x60000fe36180),                              found btree_range_pin_t(begin=0, end=18446744073709551615, depth=3, extent=0x600                             01766ed80)
ERROR 2021-08-27 22:49:31,144 [shard 0] none - ../src/crimson/os/seastore/lba_ma                             nager/btree/btree_range_pin.cc:116 : In function 'void crimson::os::seastore::lb                             a_manager::btree::btree_pin_set_t::add_pin(crimson::os::seastore::lba_manager::b                             tree::btree_range_pin_t&)', ceph_assert(%s)
0 == "impossible"
Aborting on shard 0.
Backtrace:
 0# 0x000056109FFE1ECF in /home/chunmei/ceph/build/bin/crimson-osd
 1# FatalSignal::signaled(int, siginfo_t const&) in /home/chunmei/ceph/build/bin                             /crimson-osd
 2# FatalSignal::install_oneshot_signal_handler<6>()::{lambda(int, siginfo_t*, v                             oid*)#1}::_FUN(int, siginfo_t*, void*) in /home/chunmei/ceph/build/bin/crimson-o                             sd
 3# 0x00007F5D8CA273C0 in /lib/x86_64-linux-gnu/libpthread.so.0
 4# gsignal in /lib/x86_64-linux-gnu/libc.so.6
 5# abort in /lib/x86_64-linux-gnu/libc.so.6
 6# ceph::__ceph_assert_fail(char const*, char const*, int, char const*) in /hom                             e/chunmei/ceph/build/bin/crimson-osd
 7# ceph::__ceph_assertf_fail(char const*, char const*, int, char const*, char c                             onst*, ...) in /home/chunmei/ceph/build/bin/crimson-osd
 8# crimson::os::seastore::lba_manager::btree::btree_pin_set_t::add_pin(crimson:                             :os::seastore::lba_manager::btree::btree_range_pin_t&) in /home/chunmei/ceph/bui                             ld/bin/crimson-osd

**Comment by athanatos** [comment]
Likely a different cause -- anything that can cause duplicate LBA pins will cause this crash.  Will try to reproduce.

**Comment by liu-chunmei** [comment]
metrics profiling: https://gist.github.com/liu-chunmei/f2a1960ed096a6fbba4826fb56953f43
 and https://gist.github.com/liu-chunmei/45c56b0d1dcc27172f9c03fc06476035

**Pull request merged**